sphinx/ext/napoleon/docstring.py

```diff
@@ -846,8 +848,13 @@ def combine_set(tokens):
 
 def _tokenize_type_spec(spec: str) -> List[str]:
     def postprocess(item):
-        if item.startswith("default"):
-            return [item[:7], item[7:]]
+        if item.startswith("default") and item != "default":
```

Comment:
I noticed this would match "defaultdict" unexpectedly. How about using `item.startswith("default ")` (append a space after "default") instead?

Reply:
oh, yes, I didn't consider that. However, when attempting to add the default syntax to the `numpydoc` format guide someone reminded me that there are lots of projects with different formats, so right now we're trying to come up with a way to resolve this. I think for now it would be good not to restrict the syntax too much. How about something like
```suggestion
        if re.match(r"^default[^_0-9a-zA-Z].*$", item):
```

Reply:
As I commented on the last PR, we should not enhance numpydoc itself. We don't need to support dialects of numpydoc on napoleon. I don't object to choose an unrestricted way. But it does not mean supporting them all.

Reply:
if you look at numpy/numpydoc#289, it seems the format for `default` is not standardized at all, so we're considering mentioning three of them (`default x`, `default: x` and `default=x`) and state that there might be more. I was thinking of explicitly supporting those three.

Reply:
No reason to support non-standardized notations. Let see what happens in https://github.com/numpy/numpydoc/pull/289.
It would be better to split a PR to around "default" and others.

Reply:
fair enough. This might be something that should be discussed as part of that "default" PR, but what do you think about removing `: ` from `_token_regex` (which I think we shouldn't support if not directly after `default`) and use something like:
```python
pattern = re.compile(
    r"^"
    r"(default)"
    r"([^_0-9A-Za-z]+)"
    rf"({_xref_regex}|(?:[_A-Za-z][_A-Za-z0-9]*))?"
    r"$"
)
match = pattern.match(item)
if match is not None:
    return [_ for _ in match.groups() if _ is not None]
else:
    return [item]
```
in the postprocessing step? That way we "accidentally" support all the `default<delimiter><obj>` notations but can still decide to only test a limited set of different notations.

Reply:
In my experience, users who depend on such "accidental" features sometimes report a bug when it will be lost in the unexpected change.

Reply:
agreed. I guess we should continue this discussion once the `numpydoc` format guide PR was merged.

Reply:
I'll accept the change with pleasure after numpydoc guide updated :-)